pkg/lockfile/parse-maven-lock.go

```diff
@@ -68,6 +68,28 @@ type MavenLockFile struct {
 	ManagedDependencies []MavenLockDependency `xml:"dependencyManagement>dependencies>dependency"`
 }
 
+func (mlf *MavenLockFile) collectPackages(dependencies []MavenLockDependency) map[string]PackageDetails {
```

Comment:
maybe add comments stating that our implementation follows Maven's documentation but it is different from how `mvn` actually does

Reply:
I think we should follow `mvn`s rules over documentation, and reflect real world behaviour (and document this).

Reply:
I did this before I knew that `mvn` doesn't actually follow this - I think that's a far bigger thing that should be focused on first; if `mvn` doesn't follow its own spec here then I think 1. that's a huge bug in `mvn` (or its docs) which someone should engage with the community to get at least clarified and 2. we should probably just say the current implementation is fine as I imagine it's more realistic to be doing what `mvn` actually _does_ rather than what it says